### PR TITLE
Update asset reference to describe single and multiple build assets

### LINKED ISSUE #1685: Document new builds attribute in assets
Work on this documentation should not start until https://github.com/sensu/sensu-go/pull/3222 is merged. I will update this issue if any changes to the specification occur.

We're adding support for a new `builds` attribute within assets. The attribute will contain a list of zero or more builds which must contain:
* `URL` - the URL to the build-specific asset (e.g. linux/amd64)
* `Sha512` - the checksum for the build-specific asset
* `Filters` - a list of zero or more filter strings (example below)

For compiled assets, the filters will typically check for two or three fields, for example:

```
- entity.system.os == 'linux'
- entity.system.arch == 'arm'
- entity.system.arm_version == 7
```

## This is an issue with:

- [ ] Bug (site functionality or styling)
- [ ] Errata (fix needed for doc content)
- [x] New content (guide wanted)
- [ ] Update (Add missing or refresh existing content)
- [ ] Enhancement (add new site functionality)

## Context
See https://github.com/sensu/sensu-go/pull/3222.

### PR DESCRIPTION
**Pull request by cwjohnston**
## Description

These changes update the Asset reference to cover the addition of `builds` as an attribute of asset definition.

## Motivation and Context

Closes #1685 

## Review Instructions

Please review the following new sections for clarity:

- [ ] "What are asset builds?"
- [ ] "How are asset builds evaluated?"
- [ ] "How are asset builds installed?"
- [ ] "How are asset builds executed?"

### CODE REVIEW

**Review by rgeniesse — APPROVED** [review]
LGTM 💯 .

File: content/sensu-go/5.12/reference/assets.md [hunk 1]
```diff
@@ -235,26 +301,27 @@ spec:
 
 {{< /language-toggle >}}
 
-### Asset definition
+### Asset definition (single build)
```
Comment (amdprophet) [inline content/sensu-go/5.12/reference/assets.md]: We're deprecating this single build style of assets. The recommended method is to add a single build to the `builds` array.

**Review by amdprophet — CHANGES_REQUESTED** [review]

**Comment by treydock** [comment]
I was experimenting with this using CI builds for sensu-puppet module and noticed that `headers` is something I can set for builds but didn't notice it in this pull request. Should that be called out?

**Comment by cwjohnston** [comment]
> I was experimenting with this using CI builds for sensu-puppet module and noticed that `headers` is something I can set for builds but didn't notice it in this pull request. Should that be called out?

@amdprophet can you please clarify how headers are meant to be specified for multiple builds vs single builds?

**Comment by amdprophet** [comment]
@cwjohnston builds each have their own headers field that can be used for tokens, auth, etc. that may be required to fetch from some sources (e.g. S3 bucket). The headers field will need to be specified for any builds that require it.

**Review by amdprophet — APPROVED** [review]
This looks solid. Great work!

**Pull request merged**